app/src/main/java/com/andrew/tasky/agenda/data/agenda/AgendaRepositoryImpl.kt

```diff
@@ -213,17 +212,17 @@ class AgendaRepositoryImpl(
         taskRepository.uploadCreateAndUpdateModifiedTasks()
         eventRepository.uploadCreateAndUpdateModifiedEvents()
 
-        val reminderDeleteIds = db.getReminderDao().getModifiedReminders().filter {
+        val reminderDeleteIds = db.getReminderDao().getModifiedReminders().first().filter {
             it.modifiedType == ModifiedType.DELETE
         }.map {
             it.id
         }
-        val taskDeleteIds = db.getTaskDao().getModifiedTasks().filter {
+        val taskDeleteIds = db.getTaskDao().getModifiedTasks().first().filter {
             it.modifiedType == ModifiedType.DELETE
         }.map {
             it.id
         }
-        val eventDeleteIds = db.getEventDao().getModifiedEvents().filter {
+        val eventDeleteIds = db.getEventDao().getModifiedEvents().first().filter {
```

Comment:
Here you could execute all these calls in parallel with `async` blocks. Right now it's sequential